### ISSUE TITLE
Multiple success request feature executes much more than the specified number of times

### ISSUE THREAD

**Issue by cmeeren** [luuksommers/vsts-website-warmup#19]
Previously mentioned in https://github.com/luuksommers/vsts-website-warmup/issues/13#issuecomment-1207950069 and https://github.com/luuksommers/vsts-website-warmup/issues/13#issuecomment-1210415253.

There seems to be a bug in the recent feature (referred to in https://github.com/luuksommers/vsts-website-warmup/issues/13#issuecomment-1201672943) that added support for requiring multiple consecutive success requests. It seems that it starts counting not at 1, but at 0 or a negative number (that varies from run to run), and counts up from there. This means that the task potentially requires many more successful requests than what is configured.

For all of the examples below, I have configured 3 successful requests.

Below, it starts counting at 0, and therefore actually performs 4 consecutive successful requests. See the last four `Invoke-WebRequest` below.

```
2022-08-10T09:27:15.9917421Z ##[section]Starting: Warmup website
2022-08-10T09:27:16.0203570Z ==============================================================================
2022-08-10T09:27:16.0203883Z Task         : Website Warmup
2022-08-10T09:27:16.0204117Z Description  : This task warms up a website by calling its url
2022-08-10T09:27:16.0204330Z Version      : 2.2.0
2022-08-10T09:27:16.0204505Z Author       : Luuk Sommers
2022-08-10T09:27:16.0205355Z Help         : [More Information](https://github.com/luuksommers/vsts-website-warmup)
2022-08-10T09:27:16.0205679Z ==============================================================================
2022-08-10T09:27:26.4833026Z Invoke-WebRequest https://chkboxapi-production-swap.azurewebsites.net/maintenance/health/ready, try 1 / 15
2022-08-10T09:27:27.0114269Z ##[warning]The remote server returned an error: (403) Forbidden.
2022-08-10T09:27:27.0122746Z Sleep for 10 seconds, before try 2 / 15
2022-08-10T09:27:37.0136743Z Invoke-WebRequest https://chkboxapi-production-swap.azurewebsites.net/maintenance/health/ready, try 2 / 15
2022-08-10T09:27:37.5676122Z Site is up, check 0 / 3, will do another check.
2022-08-10T09:27:47.5686552Z Invoke-WebRequest https://chkboxapi-production-swap.azurewebsites.net/maintenance/health/ready, try 3 / 15
2022-08-10T09:27:47.8542457Z ##[warning]The remote server returned an error: (503) Server Unavailable.
2022-08-10T09:27:47.8557930Z Sleep for 10 seconds, before try 4 / 15
2022-08-10T09:27:57.8572832Z Invoke-WebRequest https://chkboxapi-production-swap.azurewebsites.net/maintenance/health/ready, try 4 / 15
2022-08-10T09:28:08.6605425Z Site is up, check 0 / 3, will do another check.
2022-08-10T09:28:18.6630322Z Invoke-WebRequest https://chkboxapi-production-swap.azurewebsites.net/maintenance/health/ready, try 5 / 15
2022-08-10T09:28:19.0744945Z Site is up, check 1 / 3, will do another check.
2022-08-10T09:28:29.0762477Z Invoke-WebRequest https://chkboxapi-production-swap.azurewebsites.net/maintenance/health/ready, try 6 / 15
2022-08-10T09:28:29.4754616Z Site is up, check 2 / 3, will do another check.
2022-08-10T09:28:39.4778868Z Invoke-WebRequest https://chkboxapi-production-swap.azurewebsites.net/maintenance/health/ready, try 7 / 15
2022-08-10T09:28:39.8635154Z Site is running in 73.3869983 seconds
2022-08-10T09:28:39.9262744Z ##[section]Finishing: Warmup website
```

Here is another similar one that starts counting at -1, and therefore performs 5 requests in total:

```
2022-08-19T10:43:52.3635313Z ##[section]Starting: Warmup website
2022-08-19T10:43:52.3762249Z ==============================================================================
2022-08-19T10:43:52.3762537Z Task         : Website Warmup
2022-08-19T10:43:52.3762787Z Description  : This task warms up a website by calling its url
2022-08-19T10:43:52.3763006Z Version      : 2.2.0
2022-08-19T10:43:52.3763187Z Author       : Luuk Sommers
2022-08-19T10:43:52.3764252Z Help         : [More Information](https://github.com/luuksommers/vsts-website-warmup)
2022-08-19T10:43:52.3764780Z ==============================================================================
2022-08-19T10:43:53.5619782Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 1 / 15
2022-08-19T10:43:53.8333313Z ##[warning]The remote server returned an error: (403) Forbidden.
2022-08-19T10:43:53.8346170Z Sleep for 10 seconds, before try 2 / 15
2022-08-19T10:44:03.8368770Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 2 / 15
2022-08-19T10:44:03.9572196Z ##[warning]The remote server returned an error: (403) Forbidden.
2022-08-19T10:44:03.9586009Z Sleep for 10 seconds, before try 3 / 15
2022-08-19T10:44:13.9557368Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 3 / 15
2022-08-19T10:44:14.4580288Z Site is up, check -1 / 3, will do another check.
2022-08-19T10:44:24.4559789Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 4 / 15
2022-08-19T10:44:24.4888088Z ##[warning]The remote server returned an error: (503) Server Unavailable.
2022-08-19T10:44:24.4900082Z Sleep for 10 seconds, before try 5 / 15
2022-08-19T10:44:34.4963686Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 5 / 15
2022-08-19T10:44:34.6817930Z Site is up, check -1 / 3, will do another check.
2022-08-19T10:44:44.6857212Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 6 / 15
2022-08-19T10:44:44.8532624Z Site is up, check 0 / 3, will do another check.
2022-08-19T10:44:54.8463187Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 7 / 15
2022-08-19T10:44:55.0085428Z Site is up, check 1 / 3, will do another check.
2022-08-19T10:45:05.0114602Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 8 / 15
2022-08-19T10:45:05.1678354Z Site is up, check 2 / 3, will do another check.
2022-08-19T10:45:15.1696103Z Invoke-WebRequest https://app-vbit-api-attachment-prod-002-swap.azurewebsites.net/maintenance/health/ready, try 9 / 15
2022-08-19T10:45:15.3206588Z Site is running in 81.7673991 seconds
2022-08-19T10:45:15.3424086Z ##[section]Finishing: Warmup website
```

Here is one that starts at -3, and therefore requires 7 consecutive successful requests in total:

```
2022-08-08T09:41:02.2244697Z ##[section]Starting: Warmup website
2022-08-08T09:41:02.2363044Z ==============================================================================
2022-08-08T09:41:02.2363367Z Task         : Website Warmup
2022-08-08T09:41:02.2363612Z Description  : This task warms up a website by calling its url
2022-08-08T09:41:02.2363853Z Version      : 2.2.0
2022-08-08T09:41:02.2364152Z Author       : Luuk Sommers
2022-08-08T09:41:02.2364985Z Help         : [More Information](https://github.com/luuksommers/vsts-website-warmup)
2022-08-08T09:41:02.2365347Z ==============================================================================
2022-08-08T09:41:05.6374238Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 1 / 10
2022-08-08T09:41:06.2362151Z ##[warning]The remote server returned an error: (403) Forbidden.
2022-08-08T09:41:06.2377693Z Sleep for 10 seconds, before try 2 / 10
2022-08-08T09:41:16.2398283Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 2 / 10
2022-08-08T09:41:16.3900766Z ##[warning]The remote server returned an error: (503) Server Unavailable.
2022-08-08T09:41:16.3907291Z Sleep for 10 seconds, before try 3 / 10
2022-08-08T09:41:26.3925715Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 3 / 10
2022-08-08T09:41:26.4331317Z ##[warning]The remote server returned an error: (503) Server Unavailable.
2022-08-08T09:41:26.4342467Z Sleep for 10 seconds, before try 4 / 10
2022-08-08T09:41:36.4362724Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 4 / 10
2022-08-08T09:43:40.0438304Z ##[warning]The remote server returned an error: (503) Server Unavailable.
2022-08-08T09:43:40.0444623Z Sleep for 10 seconds, before try 5 / 10
2022-08-08T09:43:50.0458284Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 5 / 10
2022-08-08T09:45:43.0124552Z Site is up, check -3 / 3, will do another check.
2022-08-08T09:45:53.0132286Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 6 / 10
2022-08-08T09:45:53.1802277Z Site is up, check -2 / 3, will do another check.
2022-08-08T09:46:03.1816828Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 7 / 10
2022-08-08T09:46:03.3453742Z Site is up, check -1 / 3, will do another check.
2022-08-08T09:46:13.3472068Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 8 / 10
2022-08-08T09:46:13.6955004Z Site is up, check 0 / 3, will do another check.
2022-08-08T09:46:23.6977390Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 9 / 10
2022-08-08T09:46:23.8502033Z Site is up, check 1 / 3, will do another check.
2022-08-08T09:46:34.2991333Z Invoke-WebRequest https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready, try 10 / 10
2022-08-08T09:46:34.2992183Z Site is up, check 2 / 3, will do another check.
2022-08-08T09:46:44.0102306Z Site returned error after 10 tries and in 338.376453 seconds!
2022-08-08T09:46:44.1858270Z ##[error]Error warm-up https://app-vbit-api-authorization-dev-002-swap.azurewebsites.net/maintenance/health/ready
2022-08-08T09:46:44.2130578Z ##[section]Finishing: Warmup website
```

**Comment by luuksommers** [comment]
Bugfix is released in the marketplace, version 1.2.1

**Comment by cmeeren** [comment]
Great, thanks! 🥳
